### ISSUE TITLE
[Feature Request] Add cancel button in the Transform task UI

### ISSUE THREAD

**Issue by rahmanshaber** [FreeCAD/FreeCAD#7517]
### Is there an existing request for this?

- [X] I have searched the existing requests

### Forums discussion

https://forum.freecadweb.org/viewtopic.php?f=34&t=71598

### Idea description

Sometime we just move a part to compare with another, but to go back to previous position we need to click OK and then undo the changes.
But if the part is filled with details then it takes time to undo the changes. So with a "Cancel" button this can be prevented.
Also if you move the part in 6 step than you have to press ctrl+z 6 times to go to previous position.

With the cancel button that will be reduced and the UI consistency will also be maintained. 

![image](https://user-images.githubusercontent.com/30831735/191811014-50cda561-e279-45a4-a807-9164c3a32785.png)

Requested UI 
![image](https://user-images.githubusercontent.com/30831735/191811050-2ebef792-1695-446f-81bb-870e96c428ec.png)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by luzpaz** [comment]
@adrianinsaval do you want to tackle this? :stuck_out_tongue_winking_eye:

**Comment by adrianinsaval** [comment]
I agree with the request but don't really know how to do it so it's gonna be a pass from me for the time being, perhaps at a later time I'll learn some about all the transaction and undo mumbo jumbo.

**Comment by luzpaz** [comment]
Per wwmayer in https://github.com/FreeCAD/FreeCAD/pull/9647#issuecomment-1560690244

> Just by adding a Cancel button the issue is not solved because when clicking the button after a transformation the geometry doesn't jump back.
> 
> What's still needed is:
> 
>     * the transaction handling must be moved from ViewProviderDragger (dragStartCallback/dragFinishCallback) to TaskCSysDragger  
>     * TaskCSysDragger::accept() must commit the pending transaction
>     * implement TaskCSysDragger::reject() that just aborts a pending transaction

**Comment by Aryan01b** [comment]
@luzpaz, Can i work on this issue.
if so, then Which file should be modified to add the feature?

**Comment by luzpaz** [comment]
@Aryan01b perhaps @PaddleStroke can give some insight ?

**Comment by PaddleStroke** [comment]
Same as Adrian here, I don't know the transaction system.
But wwmayer has already said what needs to be done in the message you
quoted no?

**Comment by luzpaz** [comment]
Thanks for responding @PaddleStroke, just for clarification, I was quoting wwmayer from a closed PR (not a direct message). It's not clear what file(s) he was referring to specifically.

**Comment by didendron** [comment]
@luzpaz, Could I pick this as my first issue?

**Comment by luzpaz** [comment]
@didendron go for it!
